V1.0.0

cat-haines opened the pull request:
- Fully namespaced constants
- Changed behaviour of enable/disable/size to bring inline with hardware.spiflash
- Significantly refactored _preverify and _postverify methods (where the most processing takes place)

The following table is the performance for 3x8K writes:

| Task | Original | First Refactor | Final Version | hardware.spiflash (reference |
| --- | --- | --- | --- | --- |
| no verification | 117ms | 92ms | 92ms | 110ms |
| SPIFLASH_PREVERIFY + SPIFLASH_POSTVERIFY | 4392ms | 1243ms | 481ms | 320ms |
| SPIFLASH_PREVERIFY | 2679ms | 666ms | 458ms | 181ms |
| SPIFLASH_POSTVERIFY | 2319ms | 665ms | 129ms | 183ms |

cat-haines committed:
Moved statics to fully namespaced consts
Changed a couple consts to strings so no .tochar is required

cat-haines committed:
Removed _getStatus method and incorporated into _wrenable and _waitForStatus
- _wrenable and _waitForStatus are now able to set _cs_l_w once instead of once per loop
- _wrenable and _waitForStatus don't call any other class methods now

cat-haines committed:
Removed _cs_l_w from _wrenable and _waitForStatus
- They now assume _cs_l_w(0) has been called previous to method
- They now assume _cs_l_w(1) will be called after the method complete

cat-haines committed:
Updated one more const that I missed

cat-haines committed:
Revered _cs_l_w behaviour.. didn't work as expected (or at all)
Refactored _preverify to read 4byte integers when possible
 - Significantly decreases compute time

cat-haines committed:
Rewrote _preverify and _postverify again
- Further reduced processing time:

| Task                                     | Original | First Refactor | This Refactor | hardware.spiflash (reference |
| --- | --- | --- | --- | --- |
| no verification                          | 117ms  | 92ms   | 92ms  | 110ms |
| SPIFLASH_PREVERIFY + SPIFLASH_POSTVERIFY | 4392ms | 1243ms | 481ms | 320ms |
| SPIFLASH_PREVERIFY                       | 2679ms | 666ms  | 458ms | 181ms |
| SPIFLASH_POSTVERIFY                      | 2319ms | 665ms  | 129ms | 183ms |

cat-haines committed:
Fixed comments + made last of consts into strings

cat-haines committed:
not including account name in consts

cat-haines committed:
One more const update to fix some bugs I introduced :(

bbharris commented on SpiFlash.class.nut:
```diff
@@ -1,68 +1,61 @@
 
+    // aliased functions to speed things up
     _cs_l_w = null;
     _spi_w = null;
```
Is this used anywhere?

bbharris replied:
Doesn't seem to be initialized anywhere. How is this working at all?

cat-haines committed:
Commited README

cat-haines committed:
Fixed link

cat-haines committed:
Italicized dataRate_kHz for configure

bbharris commented on SpiFlash.class.nut:
```diff
@@ -70,106 +63,101 @@ class SPIFlash {
-        _spi_w(RDP.tochar());
+        _spi_w(ELECTRICIMP_SPIFLASH_RDP);
         _cs_l_w(1);
     }
```
I think you have to give the SPI some time to initialize after you enable before you can issue a command. Might want to throw a little blocking sleep in here to enforce that.

cat-haines commented:
@bbharris - I think you're looking at diff files, not the actual code files.. so bits probably appear like they're missing (because GitHub doesn't always show the entire file).

cat-haines committed:
Added sleep after enable

cat-haines committed:
Minor modifications
 - checking status register in _write before calling _waitForStatus.. saves ~3ms in test.. so why not :)
 - Removed a throw statement from preverify check that was for testing and not production

cat-haines committed:
Replaced imp.sleep with status check in enable method
- Fixed timeout check in _wrenable and _waitForStatus

cat-haines commented:
Any other comments on this PR, or are we happy to merge?

cat-haines committed:
changed case of file

blindman2k commented:
I would need to test it before I could say it was ready but I am happy with what we have so far. Any issues could come through as bug reports.

blindman2k commented:
I just ran my battery of tests on this and it performs identically to the hardware.spiflash. My name is aron and I approve this message.
